core/src/main/java/me/mykindos/betterpvp/core/client/punishments/Punishment.java

```diff
@@ -60,7 +61,7 @@ public Component getPunishmentInformation() {
         Component component = Component.empty().append(currentComp).appendSpace()
                 .append(Component.text(this.type.getName(), NamedTextColor.WHITE)).appendSpace()
                 .append(Component.text(reason == null ? "No Reason" : reason, NamedTextColor.GRAY)).appendSpace()
-                .append(Component.text(punisher == null ? "SERVER" : punisher, NamedTextColor.AQUA));
+                .append(Component.text(punisher == null ? "SERVER" : Bukkit.getOfflinePlayer(UUID.fromString(punisher)).getName(), NamedTextColor.AQUA));
```

Comment:
Is this ugly? Yes. Should it be done a different way? Also yes. Hence the temporary change